src/wl_window.c

```diff
@@ -499,13 +528,22 @@ void _glfwPlatformMaximizeWindow(_GLFWwindow* window)
 
 void _glfwPlatformShowWindow(_GLFWwindow* window)
 {
-    wl_shell_surface_set_toplevel(window->wl.shell_surface);
+    if (!window->monitor)
```

Comment:
I know this was broken before, but shouldn't we also create the shell surface even if window->monitor != NULL?